quic/server/handshake/ServerHandshake.cpp

```diff
@@ -350,132 +298,183 @@ void ServerHandshake::processPendingEvents() {
-        errMsg.toStdString(),
-        static_cast<TransportErrorCode>(
-            fizz::AlertDescription::internal_error)));
+        "Unexpected close notify received",
+        TransportErrorCode::INTERNAL_ERROR));
   }
-}
 
-void ServerHandshake::ActionMoveVisitor::operator()(fizz::WaitForData&) {
-  server_.waitForData_ = true;
-}
+  void operator()(fizz::SecretAvailable& secretAvailable) {
+    auto aead = fizz::Protocol::deriveRecordAeadWithLabel(
+        *server_.state_.context()->getFactory(),
+        *server_.state_.keyScheduler(),
+        *server_.state_.cipher(),
+        folly::range(secretAvailable.secret.secret),
+        kQuicKeyLabel,
+        kQuicIVLabel);
+    auto headerCipher = server_.cryptoFactory_->makePacketNumberCipher(
+        folly::range(secretAvailable.secret.secret));
+    switch (secretAvailable.secret.type.type()) {
+      case fizz::SecretType::Type::EarlySecrets_E:
+        switch (*secretAvailable.secret.type.asEarlySecrets()) {
+          case fizz::EarlySecrets::ClientEarlyTraffic:
+            server_.zeroRttReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.zeroRttReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          default:
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::HandshakeSecrets_E:
+        switch (*secretAvailable.secret.type.asHandshakeSecrets()) {
+          case fizz::HandshakeSecrets::ClientHandshakeTraffic:
+            server_.handshakeReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.handshakeReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          case fizz::HandshakeSecrets::ServerHandshakeTraffic:
+            server_.handshakeWriteCipher_ = FizzAead::wrap(std::move(aead));
+            server_.handshakeWriteHeaderCipher_ = std::move(headerCipher);
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::AppTrafficSecrets_E:
+        switch (*secretAvailable.secret.type.asAppTrafficSecrets()) {
+          case fizz::AppTrafficSecrets::ClientAppTraffic:
+            server_.oneRttReadCipher_ = FizzAead::wrap(std::move(aead));
+            server_.oneRttReadHeaderCipher_ = std::move(headerCipher);
+            break;
+          case fizz::AppTrafficSecrets::ServerAppTraffic:
+            server_.oneRttWriteCipher_ = FizzAead::wrap(std::move(aead));
+            server_.oneRttWriteHeaderCipher_ = std::move(headerCipher);
+            break;
+        }
+        break;
+      case fizz::SecretType::Type::MasterSecrets_E:
+        break;
+    }
+    server_.handshakeEventAvailable_ = true;
+  }
 
-void ServerHandshake::ActionMoveVisitor::operator()(
-    fizz::server::MutateState& mutator) {
-  mutator(server_.state_);
-}
+ private:
+  ServerHandshake& server_;
+};
 
-void ServerHandshake::ActionMoveVisitor::operator()(
-    fizz::server::AttemptVersionFallback&) {
-  CHECK(false) << "Fallback Unexpected";
-}
+void ServerHandshake::processActions(
```

Comment:
was this changed at all or was it simpled moved to this location?

Reply:
It was simply moved. It is using ActionMoveVisitor and therefore must be defined after. The code is identical at on small exception.